tsdb/tsdbblockutil.go

```diff
@@ -61,30 +39,35 @@ func CreateBlock(samples []*MetricSample, dir string, mint, maxt int64, logger l
 	if chunkRange < 0 {
 		return "", ErrInvalidTimes
 	}
-	chunkDir := filepath.Join(dir, "chunks_tmp")
-	defer func() {
-		os.RemoveAll(chunkDir)
-	}()
-	head, err := CreateHead(samples, chunkRange, chunkDir, logger)
+
+	w := NewTSDBWriter(logger, dir)
+	err := w.initHead(chunkRange)
 	if err != nil {
 		return "", err
 	}
-	defer head.Close()
 
-	compactor, err := NewLeveledCompactor(context.Background(), nil, logger, ExponentialBlockRanges(DefaultBlockDuration, 3, 5), nil)
+	ctx := context.Background()
+	app := w.Appender(ctx)
+
+	for _, sample := range samples {
+		_, err = app.Add(sample.Labels, sample.TimestampMs, sample.Value)
+		if err != nil {
+			return "", err
+		}
+	}
+	err = app.Commit()
```

Comment:
We can club this with the below line as `if err := app.Commit(); err != nil {` (same for other places where the only return value is an error)